UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs

```diff
@@ -831,7 +849,83 @@ protected virtual IEnumerator ConstructNode(Node node, int nodeIndex)
 			_assetCache.NodeCache[nodeIndex] = nodeObj;
 		}
 
-		protected virtual IEnumerator ConstructMesh(GLTF.Schema.Mesh mesh, Transform parent, int meshId)
+		bool NeedsSkinnedMeshRenderer(MeshPrimitive primitive, Skin skin)
+		{
+			return HasBones(skin) || HasBlendShapes(primitive);
+		}
+
+		bool HasBones(Skin skin)
+		{
+			return skin != null;
+		}
+
+		bool HasBlendShapes(MeshPrimitive primitive)
+		{
+			return primitive.Targets != null;
+		}
+
+		void SetupBones(Skin skin, MeshPrimitive primitive, SkinnedMeshRenderer renderer, GameObject primitiveObj, UnityEngine.Mesh curMesh)
+		{
+			var boneCount = skin.Joints.Count;
+			Transform[] bones = new Transform[boneCount];
+
+			int bufferId = skin.InverseBindMatrices.Value.BufferView.Value.Buffer.Id;
+			BufferCacheData bufferCacheData = _assetCache.BufferCache[bufferId];
+			AttributeAccessor attributeAccessor = new AttributeAccessor
+			{
+				AccessorId = skin.InverseBindMatrices,
+				Stream = _assetCache.BufferCache[bufferId].Stream,
+				Offset = _assetCache.BufferCache[bufferId].ChunkOffset
+			};
+
+			GLTFHelpers.BuildBindPoseSamplers(ref attributeAccessor);
+
+			GLTF.Math.Matrix4x4[] gltfBindPoses = attributeAccessor.AccessorContent.AsMatrix4x4s;//.ToUnityMatrix4x4sConvert();
```

Comment:
what is commented out at the end

Reply:
Whoops - I moved that to the for loop below it and forgot to remove it. It's gone now